src/components/form-editor/modals/new-question.tsx

```diff
@@ -0,0 +1,397 @@
+import React, { useCallback, useContext, useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import {
+  Button,
+  ComboBox,
+  ComposedModal,
+  Form,
+  FormGroup,
+  ModalBody,
+  ModalFooter,
+  ModalHeader,
+  Row,
+  Select,
+  SelectItem,
+  TextInput,
+} from "carbon-components-react";
+import { Answer, Concept, ConceptMapping, Question } from "../../../api/types";
+import { Add } from "@carbon/icons-react/next";
+import { SchemaContext } from "../../../context/context";
+import { showToast } from "@openmrs/esm-framework";
+import { useConcepts } from "../../../api/concept";
+import styles from "./modals.scss";
+
+interface CreateQuestionModalProps {
+  questions: any;
+}
+
+const CreateQuestion: React.FC<CreateQuestionModalProps> = ({ questions }) => {
+  const { t } = useTranslation();
+  const { concepts } = useConcepts();
+  const { schema, setSchema } = useContext(SchemaContext);
+  const [openCreateQuestionModal, setOpenCreateQuestionModal] = useState(false);
+  const [questionLabel, setQuestionLabel] = useState("");
+  const [questionType, setQuestionType] = useState("");
+  const [questionId, setQuestionId] = useState("");
+  // Question Options
+  const [renderElement, setRenderElement] = useState("");
+  const [answers, setAnswers] = useState<Answer[]>([]);
+  const [max, setMax] = useState("");
+  const [min, setMin] = useState("");
+  const [concept, setConcept] = useState<string>(null);
+  const [conceptMappings, setConceptMappings] = useState<ConceptMapping[]>([]);
+  const [weekList, setWeekList] = useState<any>([]);
+  const [attributeType, setAttributeType] = useState("");
+  const [rows, setRows] = useState("");
+  const [orderSettingUuid, setOrderSettingUuid] = useState("");
+  const [orderType, setOrderType] = useState("");
+  const [selectableOrders, setSelectableOrders] = useState<Answer[]>([]);
+
+  const types = [
+    {
+      key: "obs",
+      value: "obs",
+    },
+    {
+      key: "obsGroup",
+      value: "obsGroup",
+    },
+    {
+      key: "testOrder",
+      value: "testOrder",
+    },
+    {
+      key: "control",
+      value: "control",
+    },
+    {
+      key: "complex-obs",
+      value: "complex-obs",
+    },
+    {
+      key: "encounterDatetime",
+      value: "encounterDatetime",
+    },
+    {
+      key: "encounterProvider",
+      value: "encounterProvider",
+    },
+    {
+      key: "encounterLocation",
+      value: "encounterLocation",
+    },
+    {
+      key: "personAttribute",
+      value: "personAttribute",
+    },
+  ];
+  const renderElements = [
```

Comment:
Manage this state through a config. I should be able to add custom types without changing the code.